modules/story/service.js

```diff
@@ -0,0 +1,57 @@
+import Story from '../../models/Story.js'
+import View from '../../models/View.js'
+import User from '../../models/User.js'
+import mongoose from 'mongoose'
+import { ApiError } from '../../errors/ApiError.js'
+
+export const create = async ({ content, image, user_id }) => {
+    let story
+    const session = await mongoose.startSession()
+    await session.withTransaction(async () => {
+        let fullname = await User.findById([user_id], null, { session }).select('fullName')
+        let user = { user_id, fullName: fullname.fullName }
+        story = await Story.create([{ content, image, user }], { session })
+    })
+    session.endSession()
+    return story
+}
+
+export const searChTextOrUser = async (toSearch, skip) => {
+    const stories = await Story.find({ $text: { $search: toSearch } }, { score: { $meta: "textScore" } })
+        .sort({ score: { $meta: 'textScore' } })
+        .skip(skip)
+        .limit(10)
+    return stories
+}
+
+
+export const viewStory = async ({ story_id, user_id }) => {
+    let view
+    const session = await mongoose.startSession()
+    await session.withTransaction(async () => {
+        view = await View.create([{ story_id, user_id }], { session })
+        if (!view) {
+            throw ApiError.badRequest('error adding view')
+        }
+        await Story.updateOne({ _id: story_id }, { $inc: { views: 1 } }, { session })
```

Comment:
there is no guarantee that the story is there, you could solve it by making sure the update is done successfully.

```suggestion
        const result = await Story.updateOne({ _id: story_id }, { $inc: { views: 1 } }, { session })
        if(result.nModified !== 1) {
             // throw an error (abort)
        }
```